packages/core/test/blockManipulation.test.ts

```diff
@@ -1,81 +1,111 @@
-};
-
-const multipleBlocks: PartialBlock[] = [
-  {
-    type: "heading",
-    props: {
-      level: "1",
-    },
-    content: "Heading 1",
-    children: [
-      {
-        type: "heading",
-        props: {
-          level: "1",
-        },
-        content: "Nested Heading 1",
+import { Block, BlockNoteEditor, PartialBlock } from "../src";
+
+let editor: BlockNoteEditor;
+let ready = false;
+function waitForEditor() {
+  const poll = (resolve) => {
+    if(ready) {
+      resolve();
+    } else {
+      setTimeout(() => poll(resolve));
+    }
+  }
+
+  return new Promise<void>(poll);
+}
+
+let singleBlock: PartialBlock;
+
+let multipleBlocks: PartialBlock[];
+
+let insert: (placement: "before" | "nested" | "after") => Block[];
+
+beforeEach(() => {
+  (window as Window & { __TEST_OPTIONS?: {} }).__TEST_OPTIONS = {};
+
+  editor = new BlockNoteEditor({
+    onCreate: () => {
+      ready = true;
+    }
+  });
+
+  singleBlock = {
+    type: "paragraph",
+    content: "Paragraph",
+  };
+
+  multipleBlocks = [
+    {
+      type: "heading",
+      props: {
+        level: "1",
       },
-    ],
-  },
-  {
-    type: "heading",
-    props: {
-      level: "2",
-    },
-    content: "Heading 2",
-    children: [
-      {
-        type: "heading",
-        props: {
-          level: "2",
+      content: "Heading 1",
+      children: [
+        {
+          type: "heading",
+          props: {
+            level: "1",
+          },
+          content: "Nested Heading 1",
         },
-        content: "Nested Heading 2",
+      ],
+    },
+    {
+      type: "heading",
+      props: {
+        level: "2",
       },
-    ],
-  },
-];
+      content: "Heading 2",
+      children: [
+        {
+          type: "heading",
+          props: {
+            level: "2",
+          },
+          content: "Nested Heading 2",
+        },
+      ],
+    },
+  ];
 
-beforeEach(() => {
-  (window as Window & { __TEST_OPTIONS?: {} }).__TEST_OPTIONS = {};
+  insert = (placement) => {
+    const existingBlock = editor.allBlocks[0];
+    editor.insertBlocks(multipleBlocks, existingBlock, placement);
+
+    return editor.allBlocks;
+  }
 });
 
 afterEach(() => {
+  ready = false;
+  editor._tiptapEditor.destroy()
+  editor = undefined;
+
   delete (window as Window & { __TEST_OPTIONS?: {} }).__TEST_OPTIONS;
 });
 
 describe("Inserting Blocks with Different Placements", () => {
-  function insert(placement: "before" | "nested" | "after"): Block[] {
-    const editor = new BlockNoteEditor().tiptapEditor;
-    const editorAPI = new Editor(editor);
-    (window as Window & { __TEST_OPTIONS?: {} }).__TEST_OPTIONS =
-      (window as Window & { __TEST_OPTIONS?: {} }).__TEST_OPTIONS || {};
-
-    const existingBlock = editorAPI.allBlocks[0];
-
-    editorAPI.insertBlocks(multipleBlocks, existingBlock, placement);
-
-    return editorAPI.allBlocks;
-  }
-
   it("Insert before existing block", async () => {
-    const output = insert("before");
+    await waitForEditor()
```

Comment:
why is this needed?

Reply:
When you create a new `BlockNoteEditor()`, that only starts the creation of the editor. It's not ready to be used straight away. For example, if you pass an `onCreate` function where you log smth then immediately after calling `BlockNoteEditor()`, you terminate the app, nothing will get logged.

This is basically what's happening in the tests as well - the editor isn't 100% ready when we start accessing and changing blocks. So this causes, e.g. the UniqueID plugin to not be initialized yet and we see blocks with null IDs.

`waitForEditor()` waits until the editor is actually ready, then the tests continue as normal.